src/drivers/lego/driver.rs

```diff
@@ -0,0 +1,628 @@
+use std::collections::HashSet;
+use std::{error::Error, ffi::CString};
+
+use hidapi::HidDevice;
+use packed_struct::PackedStruct;
+use tokio::time::Instant;
+
+use crate::input::capability::Source;
+use crate::udev::device::UdevDevice;
+use crate::{
+    drivers::lego::{
+        event::{ImuAxisInput, TouchAxisInput, TouchButtonEvent},
+        hid_report::GamepadMode,
+        CLICK_DELAY, PAD_FORCE_NORMAL, RELEASE_DELAY, XINPUT_COMMAND_ID,
+    },
+    input::capability::Capability,
```

Comment:
Same thing here, we should try not to rely on InputPlumber data types inside the driver implementations.